app/src/main/java/it/quip/android/actvitiy/CreateQuipActivity.java

```diff
@@ -0,0 +1,104 @@
+package it.quip.android.actvitiy;
+
+
+import android.os.Bundle;
+import android.support.v4.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
+
+import java.util.List;
+
+import it.quip.android.QuipitApplication;
+import it.quip.android.R;
+import it.quip.android.fragment.CreateQuipCircleSelectFragment;
+import it.quip.android.fragment.CreateQuipComposeFragment;
+import it.quip.android.model.Circle;
+import it.quip.android.model.Quip;
+import it.quip.android.model.User;
+import it.quip.android.util.TimeUtils;
+
+public class CreateQuipActivity
+        extends AppCompatActivity
+        implements CreateQuipComposeFragment.OnComposeQuipListener {
+
+    private static final String COMPOSE = "compose_fragment";
+
+    private CreateQuipComposeFragment mCreateQuipComposeFragment;
+    private CreateQuipCircleSelectFragment mCreateQuipCircleSelectFragment;
+    private Quip mQuip;
+    private Button mBtQuip;
+
+    private void setupView() {
+        mBtQuip = (Button) findViewById(R.id.bt_quip_create_post);
+        mBtQuip.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                onPostQuip();
+            }
+        });
+    }
+
+    private void setupFragments() {
+        mCreateQuipComposeFragment = CreateQuipComposeFragment.newInstance();
+        mCreateQuipCircleSelectFragment = CreateQuipCircleSelectFragment.newInstance();
+    }
+
+    private void showComposeFragment() {
+        mCreateQuipComposeFragment.show(getSupportFragmentManager(), COMPOSE);
+    }
+
+    private void showShareFragment() {
+        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
+        ft.replace(R.id.fl_create_quip_circle_picker, mCreateQuipCircleSelectFragment);
+        ft.commit();
+    }
+
+    private void onPostQuip() {
+        if (mQuip != null) {
+            List<Circle> circles = mCreateQuipCircleSelectFragment.getSelected();
+            Long timestamp = TimeUtils.currentTimestampInS();
+
+            // TODO: Figure out how to add a "Share with all friends" option rather than not selecting any circles...
+            if (circles.size() > 0) {
+                for (Circle circle : circles) {
+                    Quip quip = new Quip(mQuip);
+                    quip.setCircle(circle);
+                    quip.setTimestamp(timestamp);
+                    quip.saveInBackground();
+                }
+            } else {
```

Comment:
So right now, I haven't come up with a good way to share it with your generic "all friends" circle. For now, I have it send to that if you haven't selected any circle. See the TODO message above.